{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# RNA-seq quantification"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!gdown --folder https://drive.google.com/drive/folders/1YN4FEJdWyKzPQziyFFCE7iyipXsVERwp?usp=drive_link"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!wget http://hgdownload.cse.ucsc.edu/goldenPath/sacCer3/bigZips/genes/sacCer3.ensGene.gtf.gz -O data/sacCer3.ensGene.gtf.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m    _____ _    _ ____  _____  ______          _____  \n",
      "       \u001b[44;37m =====      \u001b[0m\u001b[36m   / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "       \u001b[44;37m   =====    \u001b[0m\u001b[36m  | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "       \u001b[44;37m     ====   \u001b[0m\u001b[36m   \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "       \u001b[44;37m       ==== \u001b[0m\u001b[36m   ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "       \u001b[44;37m ========== \u001b[0m\u001b[36m  |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\u001b[0m\n",
      "\t  v2.0.8\n",
      "\n",
      "//==========================\u001b[36m featureCounts setting \u001b[0m===========================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Input files : \u001b[36m6 BAM files  \u001b[0m \u001b[0m                                   ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||                           \u001b[36mWT_C_1.bam\u001b[0m \u001b[0m                                      ||\n",
      "||                           \u001b[36mWT_C_2.bam\u001b[0m \u001b[0m                                      ||\n",
      "||                           \u001b[36mWT_C_3.bam\u001b[0m \u001b[0m                                      ||\n",
      "||                           \u001b[36mWT_E_1.bam\u001b[0m \u001b[0m                                      ||\n",
      "||                           \u001b[36mWT_E_2.bam\u001b[0m \u001b[0m                                      ||\n",
      "||                           \u001b[36mWT_E_3.bam\u001b[0m \u001b[0m                                      ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||             Output file : \u001b[36msacCer.featureCounts.tsv\u001b[0m \u001b[0m                        ||\n",
      "||                 Summary : \u001b[36msacCer.featureCounts.tsv.summary\u001b[0m \u001b[0m                ||\n",
      "||              Paired-end : \u001b[36myes\u001b[0m \u001b[0m                                             ||\n",
      "||        Count read pairs : \u001b[36mno\u001b[0m \u001b[0m                                              ||\n",
      "||              Annotation : \u001b[36msacCer3.ensGene.gtf (GTF)\u001b[0m \u001b[0m                       ||\n",
      "||      Dir for temp files : \u001b[36mdata\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "||                 Threads : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||                   Level : \u001b[36mmeta-feature level\u001b[0m \u001b[0m                              ||\n",
      "||      Multimapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "|| Multi-overlapping reads : \u001b[36mnot counted\u001b[0m \u001b[0m                                     ||\n",
      "||   Min overlapping bases : \u001b[36m1\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n",
      "//=================================\u001b[36m Running \u001b[0m==================================\\\\\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Load annotation file sacCer3.ensGene.gtf \u001b[0m... \u001b[0m                              ||\n",
      "||    Features : \u001b[36m7507\u001b[0m \u001b[0m                                                        ||\n",
      "||    Meta-features : \u001b[36m7127\u001b[0m \u001b[0m                                                   ||\n",
      "||    Chromosomes/contigs : \u001b[36m17\u001b[0m \u001b[0m                                               ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file WT_C_1.bam... \u001b[0m                                            ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
      "||    Total alignments : \u001b[36m13752146\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m10641971 (77.4%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.10 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file WT_C_2.bam... \u001b[0m                                            ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
      "||    Total alignments : \u001b[36m13355134\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m10188169 (76.3%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.09 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file WT_C_3.bam... \u001b[0m                                            ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
      "||    Total alignments : \u001b[36m14192092\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m10779709 (76.0%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.10 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file WT_E_1.bam... \u001b[0m                                            ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
      "||    Total alignments : \u001b[36m15311394\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m12184691 (79.6%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.11 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file WT_E_2.bam... \u001b[0m                                            ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
      "||    Total alignments : \u001b[36m14962268\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m11871408 (79.3%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.10 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Process BAM file WT_E_3.bam... \u001b[0m                                            ||\n",
      "||    Paired-end reads are included. \u001b[0m                                         ||\n",
      "||    The reads are assigned on the single-end mode. \u001b[0m                         ||\n",
      "||    Total alignments : \u001b[36m15151398\u001b[0m \u001b[0m                                            ||\n",
      "||    Successfully assigned alignments : \u001b[36m12156736 (80.2%)\u001b[0m \u001b[0m                    ||\n",
      "||    Running time : \u001b[36m0.10 minutes\u001b[0m \u001b[0m                                            ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Write the final count table. \u001b[0m                                              ||\n",
      "|| Write the read assignment summary. \u001b[0m                                        ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "|| Summary of counting results can be found in file \"data/sacCer.featureCoun \u001b[0m ||\n",
      "|| ts.tsv.summary\" \u001b[0m                                                           ||\n",
      "||  \u001b[0m                                                                          ||\n",
      "\\\\============================================================================//\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!featureCounts \\\n",
    "-p \\\n",
    "-a data/sacCer3.ensGene.gtf \\\n",
    "-o data/sacCer.featureCounts.tsv \\\n",
    "bam/WT_C_1.bam \\\n",
    "bam/WT_C_2.bam \\\n",
    "bam/WT_C_3.bam \\\n",
    "bam/WT_E_1.bam \\\n",
    "bam/WT_E_2.bam \\\n",
    "bam/WT_E_3.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Convert to count matrix\n",
    "This command is essentially creating a simplified count matrix by:\n",
    "1. Extracting only necessary columns (gene IDs and counts)\n",
    "2. Cleaning up sample names\n",
    "3. Reformatting to a tab-delimited file\n",
    "4. Making it suitable for downstream analysis in R/DESeq2\n",
    "\n",
    "Details explanation\n",
    "1. `cat data/sacCer.featureCounts.tsv`\n",
    "- Reads the content of the featureCounts output file\n",
    "- featureCounts typically outputs a tab-separated file with gene counts\n",
    "\n",
    "2. `awk '(NR>1) {printf \"%s \", $1; for (i=7; i<=NF; i++) printf \"%s \", $i; print \"\"}'`\n",
    "- `NR>1`: Skips the header line\n",
    "- `printf \"%s \", $1`: Prints the first column (gene IDs)\n",
    "- `for (i=7; i<=NF; i++)`: Loops through columns starting from 7th to last column\n",
    "- These columns contain the actual count values\n",
    "- First 6 columns in featureCounts output typically contain gene information (ID, Chr, Start, End, Strand, Length)\n",
    "\n",
    "3. `sed s/bam//g`\n",
    "- Removes 'bam' from sample names\n",
    "- Cleans up file names\n",
    "\n",
    "4. `tr -d \"/\"` \n",
    "- Removes forward slashes from the output\n",
    "- Further cleans file paths\n",
    "\n",
    "5. `tr -d \".\"` \n",
    "- Removes dots from the output\n",
    "- Additional filename cleaning\n",
    "\n",
    "6. `tr \" \" \"\\t\"`\n",
    "- Converts spaces to tabs\n",
    "- Makes the output tab-delimited\n",
    "\n",
    "7. `> data/sacCer_counts_raw.tsv`\n",
    "- Saves the processed output to a new file\n",
    "- Creates a clean count matrix with just gene IDs and counts\n",
    "\n",
    "Example transformation:\n",
    "```\n",
    "Original featureCounts output:\n",
    "GeneID  Chr  Start  End  Strand  Length  ./sample1.bam  ./sample2.bam\n",
    "YDL248W  chr4  1802  2953  +  1152  45  67\n",
    "\n",
    "After processing:\n",
    "YDL248W  45  67\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "!cat data/sacCer.featureCounts.tsv | awk '(NR>1) {printf \"%s \", $1; for (i=7; i<=NF; i++) printf \"%s \", $i; print \"\"}' | sed s/bam//g | tr -d \"/\" | tr -d \".\" | tr \" \" \"\\t\" > data/sacCer_counts_raw.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(7127, 8)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Geneid</th>\n",
       "      <th>WT_C_1</th>\n",
       "      <th>WT_C_2</th>\n",
       "      <th>WT_C_3</th>\n",
       "      <th>WT_E_1</th>\n",
       "      <th>WT_E_2</th>\n",
       "      <th>WT_E_3</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>YDL248W</td>\n",
       "      <td>248</td>\n",
       "      <td>224</td>\n",
       "      <td>319</td>\n",
       "      <td>523</td>\n",
       "      <td>485</td>\n",
       "      <td>586</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>YDL247W-A</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>YDL247W</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>YDL246C</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>YDL245C</td>\n",
       "      <td>22</td>\n",
       "      <td>6</td>\n",
       "      <td>10</td>\n",
       "      <td>60</td>\n",
       "      <td>34</td>\n",
       "      <td>28</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Geneid  WT_C_1  WT_C_2  WT_C_3  WT_E_1  WT_E_2  WT_E_3  Unnamed: 7\n",
       "0    YDL248W     248     224     319     523     485     586         NaN\n",
       "1  YDL247W-A       0       2      12       0       0      12         NaN\n",
       "2    YDL247W       2       0       1       0       0       5         NaN\n",
       "3    YDL246C       0       0       4       0       0      12         NaN\n",
       "4    YDL245C      22       6      10      60      34      28         NaN"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df_gene_count = pd.read_csv(\"data/sacCer_counts_raw.tsv\", sep=\"\\t\")\n",
    "print(df_gene_count.shape)\n",
    "df_gene_count.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "gene-expression-analysis",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
